### ISSUE TITLE
Support for scikit-learn's LatentDirichletAllocation and NMF, extend support for GaussianMixtureModel and BayesianGaussianMixtureModel

### ISSUE THREAD

**Issue by jeremymanning** [ContextLab/hypertools#174]
We should add support for LDA and NMF: http://scikit-learn.org/stable/modules/classes.html#module-sklearn.decomposition

These can be supported both via `cluster` and `reduce`:

Cluster: behaves similarly to GaussianMixtureModel (estimate how much of each factor is reflected by/in each observation)
Reduce: to reduce the data to n-dimensions, fit LDA/NMF with n components

Some sample code may be found here: https://github.com/ContextLab/storytelling-with-data/blob/master/data-stories/twitter-finance/twitter-finance.ipynb

We could play a similar trick with Gaussian mixture models-- to use a Gaussian mixture model to reduce the data to n dimensions, we could fit a GMM with n components and then use the membership mixing labels as the coordinates.

**Comment by jeremymanning** [comment]
Status in the 1.0 update (PR #272 https://github.com/ContextLab/hypertools/pull/272): the cluster half of this ask is fully done — `hypertools/cluster/cluster.py`'s `mixture_models` dict already supports `GaussianMixture`, `BayesianGaussianMixture`, `LatentDirichletAllocation`, and `NMF`, returning per-observation membership-proportion rows that sum to 1 (verified live for all four). The reduce half is not: `hyp.reduce(x, reduce='NMF', ndims=3)` still raises `ValueError` because `hypertools/reduce/reduce.py`'s `models` dict only has classic decomposition/manifold models.

**How this could be built on the 1.0 architecture:** `hypertools/core/model.py`'s `apply_model` already solves this — its registry merges `reduce.models`, `cluster.models`, and `cluster.mixture_models`, and `_apply_single`'s `mode='auto'` logic already picks `predict_proba` for GMM/BayesianGMM and `fit_transform` for NMF/LDA. Verified live: `apply_model(X, 'NMF', ndims=3)` and `apply_model(X, 'GaussianMixture', ndims=3, mode='predict_proba')` both work today. The concrete fix is to make `hyp.reduce()` delegate to (or share the registry/dispatch logic with) `apply_model`, rather than maintaining a separate, narrower `models` dict — either by adding the four mixture models into `reduce.models` with the same mixture-handling branch already in `cluster.py`, or by having `reduce()` fall back to `core.model.apply_model` for names it doesn't recognize locally.

**Alternatives / prior art:**
- No external library gap here — scikit-learn's own `NMF`/`LatentDirichletAllocation`/`GaussianMixture`/`BayesianGaussianMixture` are exactly what's needed and are already hypertools dependencies; this is purely a wiring gap between hypertools' own `reduce()` and `apply_model()`.
- The issue's suggestion to "fit a GMM with n components and use membership proportions as coordinates" is exactly the `mode='predict_proba'` path `apply_model` already implements.

Tagging as **low effort**; leaving open for consideration in a future release.

**Comment by jeremymanning** [comment]
Let's add mixture_models support for the reduce keyword

**Comment by jeremymanning** [comment]
Addressed in PR #272 (HyperTools 1.0), per Jeremy's directive: add mixture_models support for the `reduce` keyword (the `cluster` side — `GaussianMixture`, `BayesianGaussianMixture`, `LatentDirichletAllocation`, `NMF` — was already done).

`hyp.reduce()` now accepts all four mixture models by name and returns per-observation membership-proportion rows (each row sums to 1), exactly mirroring `hyp.cluster()`'s existing mixture output. The underlying logic (`mixture_proportions`/`normalize_membership_rows`) was factored out of `cluster.py` into shared helpers so `reduce`'s new `Reducer` base class calls the *same* functions rather than reimplementing the `predict_proba` vs. LDA/NMF row-normalization split — verified with a direct output-shape comparison test against `hyp.cluster`'s mixture path. Implemented in `hypertools/reduce/common.py` (new `REDUCERS` registry = classic reducers + the 4 mixture models) and `hypertools/reduce/reduce.py`. Commit `33fac24d`.

```python
import hypertools as hyp

proportions = hyp.reduce(x, reduce='GaussianMixture', ndims=3)
# (n, 3) array, each row sums to 1 -- soft cluster membership as a "reduction"

proportions = hyp.reduce(x, reduce='LatentDirichletAllocation', ndims=4)
```

23 new tests in `tests/test_reduce_migration.py` cover all 4 mixture models via `reduce=` (string, class, instance, dict spec), including a direct comparison against `hyp.cluster`'s mixture output for the same data, plus `return_model=True` reuse.

The README's cluster example gallery already demonstrates the soft-cluster (mixture) coloring visually:
https://raw.githubusercontent.com/ContextLab/hypertools/dev-1.0-refactor/images/cluster_example.png

Full suite: ~1275 tests passing.
